src/Couchbase.Lite.Shared/Query/QueryExpression.cs

```diff
@@ -45,14 +46,16 @@ internal abstract class QueryExpression : IExpression
 
         #region Public Methods
 
-        public static object EncodeToJSON(IList expressions)
+        public static object EncodeToJSON([NotNull]IList expressions)
         {
+            CBDebug.MustNotBeNull(WriteLog.To.Query, Tag, nameof(expressions), expressions);
```

Comment:
This is an internal class, so use `Debug.Assert` here

Reply:
This still needs to be `Debug.Assert` because it is not an outward facing method (it is `public` on an `internal` class)